pkgs/development/tools/build-managers/bazel/darwin.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/tools/cpp/osx_cc_wrapper.sh b/tools/cpp/osx_cc_wrapper.sh
+index 207bcefd78..a7b1af972a 100755
+--- a/tools/cpp/osx_cc_wrapper.sh
++++ b/tools/cpp/osx_cc_wrapper.sh
+@@ -67,10 +67,10 @@ function get_library_path() {
+ # and multi-level symlinks.
+ function get_realpath() {
+     local previous="$1"
+-    local next=$(readlink "${previous}")
++    local next=$(readlink "${previous}" | sed -e 's/\n//g')
+     while [ -n "${next}" ]; do
+         previous="${next}"
+-        next=$(readlink "${previous}")
++        next=$(readlink "${previous}" | sed -e 's/\n//g')
+     done
+     echo "${previous}"
+ }
+@@ -101,4 +101,3 @@ for rpath in ${RPATHS}; do
+         fi
+     done
+ done
+-
+diff --git a/tools/cpp/osx_cc_wrapper.sh.tpl b/tools/cpp/osx_cc_wrapper.sh.tpl
+index 4c85cd9b6b..2c49ffcfa7 100644
+--- a/tools/cpp/osx_cc_wrapper.sh.tpl
++++ b/tools/cpp/osx_cc_wrapper.sh.tpl
+@@ -69,10 +69,10 @@ function get_library_path() {
+ # and multi-level symlinks.
+ function get_realpath() {
+     local previous="$1"
+-    local next=$(readlink "${previous}")
++    local next=$(readlink "${previous}" | sed -e 's/\n//g')
+     while [ -n "${next}" ]; do
+         previous="${next}"
+-        next=$(readlink "${previous}")
++        next=$(readlink "${previous}" | sed -e 's/\n//g')
```

Comment:
Nix's `readlink` when running on macOS will output an additional newline. This does not happen with macOS' own `readlink`, or on Linux and NixOS.

Reply:
This is very weird. We are using coreutils' readlink for both Linux & macOS so I'm not sure why it is different.

Reply:
I tried to isolate the issue as I did before, but when I looked into it more I realised this is just a different symptom of the same issue as `sandbox-exec`: it's not that it's calling `readlink` differently, it's a different version of bash: see the comment about `sandbox-exec` above, and repro for env:

```
$ env ./u.sh
3.2.57(1)-release
```

Reply:
See https://github.com/NixOS/nixpkgs/pull/40990#issuecomment-392421415 .